judge/judgedaemon.main.php

```diff
@@ -314,6 +280,12 @@ private function loop(): void
             if (function_exists('pcntl_signal_dispatch')) {
                 pcntl_signal_dispatch();
             }
+            if (function_exists('pcntl_waitpid')) {
+                // Reap any finished child processes.
+                while (pcntl_waitpid(-1, $status, WNOHANG) > 0) {
+                    // Do nothing.
```

Comment:
I'd generate a warning at least if the return value is -1, when an error happened.